webextension/background.js

```diff
@@ -38,37 +38,56 @@ class StudyLifeCycleHandler {
    * call `this.enableFeature` to actually do the feature/experience/ui.
    */
   constructor() {
+    /*
+     * IMPORTANT:  Listen for `onEndStudy` before calling `browser.study.setup`
+     * because:
+     * - `setup` can end with 'ineligible' due to 'allowEnroll' key in first session.
+     *
+     */
     browser.study.onEndStudy.addListener(this.handleStudyEnding);
     browser.study.onReady.addListener(this.enableFeature);
   }
 
   /**
```

Comment:
If this is something we're going to ship, dead code like this should be removed

Reply:
By dead code you mean not used  cleanup() method, I am leaving it here in case the final shield utils v5 asks for it. I can remove it before shipping